src/components/date-time-input/date-util.ts

```diff
@@ -0,0 +1,856 @@
+  TwoDigits = '2-digit',
+}
+
+export enum DateParts {
+  Day = 'day',
+  Month = 'month',
+  Year = 'year',
+  Date = 'date',
+  Hours = 'hours',
+  Minutes = 'minutes',
+  Seconds = 'seconds',
+  AmPm = 'ampm',
+  Literal = 'literal',
+}
+
+export enum DatePart {
+  Month = 'month',
+  Year = 'year',
+  Date = 'date',
+  Hours = 'hours',
+  Minutes = 'minutes',
+  Seconds = 'seconds',
+  AmPm = 'ampm',
+}
+
+export interface DatePartInfo {
+  type: DateParts;
+  start: number;
+  end: number;
+  format: string;
+}
+
+export interface DatePartDeltas {
+  date?: number;
+  month?: number;
+  year?: number;
+  hours?: number;
+  minutes?: number;
+  seconds?: number;
+}
+
+export abstract class DateTimeUtil {
+  public static readonly DEFAULT_INPUT_FORMAT = 'MM/dd/yyyy';
+  public static readonly DEFAULT_TIME_INPUT_FORMAT = 'hh:mm tt';
+  private static readonly SEPARATOR = 'literal';
+  private static readonly DEFAULT_LOCALE = 'en';
+  private static _parser = new MaskParser();
+
+  public static parseValueFromMask(
+    inputData: string,
+    dateTimeParts: DatePartInfo[],
+    promptChar?: string
+  ): Date | null {
+    const parts: { [key in DateParts]: number } = {} as any;
+    dateTimeParts.forEach((dp) => {
+      let value = parseInt(
+        DateTimeUtil.getCleanVal(inputData, dp, promptChar),
+        10
+      );
+      if (!value) {
+        value =
+          dp.type === DateParts.Date || dp.type === DateParts.Month ? 1 : 0;
+      }
+      parts[dp.type] = value;
+    });
+    parts[DateParts.Month] -= 1;
+
+    if (parts[DateParts.Month] < 0 || 11 < parts[DateParts.Month]) {
+      return null;
+    }
+
+    // TODO: Century threshold
+    if (parts[DateParts.Year] < 50) {
+      parts[DateParts.Year] += 2000;
+    }
+
+    if (
+      parts[DateParts.Date] >
+      DateTimeUtil.daysInMonth(parts[DateParts.Year], parts[DateParts.Month])
+    ) {
+      return null;
+    }
+
+    if (
+      parts[DateParts.Hours] > 23 ||
+      parts[DateParts.Minutes] > 59 ||
+      parts[DateParts.Seconds] > 59
+    ) {
+      return null;
+    }
+
+    const amPm = dateTimeParts.find((p) => p.type === DateParts.AmPm);
+    if (amPm) {
+      parts[DateParts.Hours] %= 12;
+    }
+
+    if (
+      amPm &&
+      DateTimeUtil.getCleanVal(inputData, amPm, promptChar).toLowerCase() ===
+        'pm'
+    ) {
+      parts[DateParts.Hours] += 12;
+    }
+
+    return new Date(
+      parts[DateParts.Year] || 2000,
+      parts[DateParts.Month] || 0,
+      parts[DateParts.Date] || 1,
+      parts[DateParts.Hours] || 0,
+      parts[DateParts.Minutes] || 0,
+      parts[DateParts.Seconds] || 0
+    );
+  }
+
+  public static getDefaultMask(locale: string): string {
+    locale = locale || DateTimeUtil.DEFAULT_LOCALE;
+    if (
+      !Intl ||
+      !Intl.DateTimeFormat ||
+      !Intl.DateTimeFormat.prototype.formatToParts
+    ) {
+      // TODO: fallback with Intl.format for IE?
+      return DateTimeUtil.DEFAULT_INPUT_FORMAT;
+    }
```

Comment:
Not really needed since [our supported targets](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat#browser_compatibility) are compliant.

Reply:
probably transferred over from Angular's utils as is, but yeah sure - we can drop the check